### DISCUSSION TITLE
hover search

### DISCUSSION THREAD

**Discussion by zerx-lab** [zed-industries/zed#24625] [Feature Requests]
### Check for existing issues

- [X] Completed

### Describe the feature

![image](https://github.com/zed-industries/zed/assets/161401688/e635ff67-29aa-43b7-bf3a-15a9e1e416d7)
Can the global search be optimized to display the results in a floating manner similar to Jetbrains Idea, or to display all the results after clicking, instead of displaying them directly, and without a directory index, if there are many file results, it will be very difficult to find。
And when searching for method callers, can I use a floating window to filter, at least have a directory index, similar to vscode or idea (I think idea is more ergonomic)

**Comment by RemcoSmitsDev** [comment]
I really think that displaying it inside a panel is way better than displaying it inside a popup. It boosts my productivity in many ways. Before I was using Zed, I was using phpstorm for work, which is a pretty big project every time I have to find all references or implementations it takes a while. When the results are loaded inside the popup and I click the result that I think is the one I was looking for, I lose the ability to check the other results directly. So i have to go back and wait again before its done finding all the results which takes to much time before I finally find the result that I was looking for.

In Zed I can just click on one of the results and if its not the result I was looking for, I can just close the panel or go back to all the results and start looking again for the result that I'm looking for. Without having to wait before the language server is done sending all the results. Its not just only the interaction with the language server, but in any type of searching Zed allows you to find your result faster. 

So I guess you just have to get used to it.

**Note**: that Zed has the option to filter the results based on included/excluded files/folders. Its the filter icon right next to the search input.

**Comment by zerx-lab** [comment]
In phpstorm, this global search will pop up in the current window by default, and then click again to fix it to the bottom to complete the requirements you described. Of course, I should not compare the current progress of zed with jetbrains products. But I still hope that the experience can be better (I don't know if this is everyone's idea, but this function really likes it) or I can have a file list in the current zed search results instead of listing all the results. )

**Comment by RemcoSmitsDev** [comment]
Yeah, I know you can pin it to the bottom like a panel. But for me, It's not as powerful as the workflow in Zed out of the box. I like Zed because it is stream lined how you do things inside the editor, and it is minimal without millions of buttons.

**Comment by zerx-lab** [comment]
> Yeah, I know you can pin it to the bottom like a panel. But for me, It's not as powerful as the workflow in Zed out of the box. I like Zed because it is stream lined how you do things inside the editor, and it is minimal without millions of buttons.

But I think this is a configurable item for the user? Because the floating window is more focused on editing, there is no need to use the mouse to switch windows in focus mode. I have always liked the nvim configuration project lazyvim, because all his searches are on the current interface.

**Comment by ghost** [comment]
I would like to have an option to enable something like this, it would be much nicer to search for something in a single file without having to navigate two tabs imo

**Comment by zerx-lab** [comment]
> I would like to have an option to enable something like this, it would be much nicer to search for something in a single file without having to navigate two tabs imo

I also think this would be better, including the current goto function, which is awkward
